src/desktop/apps/authentication/helpers.ts

```diff
@@ -112,19 +119,55 @@ export const setCookies = options => {
   }
 }
 
-export const getRedirect = type => {
+export async function apiAuthWithRedirectUrl(
+  response: any,
+  redirectPath: URL
+): Promise<URL> {
+  const redirectUrl = sd.APP_URL + redirectPath.pathname
+  const accessToken = (response.user || {}).accessToken
+  const appRedirectURL = new URL(redirectUrl)
+
+  // There isn't an access token when we don't have a valid session, for example,
+  // when the user is resetting their password.
+  if (!accessToken) return appRedirectURL
+
+  try {
+    const tokenResponse = await fetch(sd.API_URL + "/api/v1/me/trust_token", {
+      method: "POST",
+      headers: { "X-Access-Token": accessToken },
+    })
+
+    if (tokenResponse.ok) {
+      const responseBody = await tokenResponse.json()
+      const trustToken = responseBody["trust_token"]
+
+      const queryParams = qs.stringify({
+        trust_token: trustToken,
+        redirect_uri: appRedirectURL.toString(),
+      })
+      return new URL(`${sd.API_URL}/users/sign_in?${queryParams}`)
+    } else {
+      return appRedirectURL
+    }
+  } catch {
```

Comment:
This will discard any errors occurring in the block above. Overall we probably don't want to affect anything in this flow if there _was_ an error occurring, so this logically accomplishes that goal. However, there would be no logging or visibility to any errors that might occur.

We can still leave this `catch`, but maybe a `raven.catpureException` call would be nice to make sure it makes it to Sentry?

Reply:
Interesting -- this just got added to the JS spec; I wonder why this works here? 

Should be `catch (error) { ... }`

Reply:
I'm curious if it's preferable here to `try`/`catch` rather than chaining on the promise and `catch` at the end?

Reply:
Weird that this worked! I'll refactor to the `catch (error) { .. }` form.

Reply:
@starsirius Can you elaborate or provide an example of what you are proposing, not really following you here.

Reply:
@dleve123 I was thinking about the typical promise handling:
```javascript
fetch(...)
  .then(...)
  .then(...)
  .catch(...)
```
But I'm not 100% sure about

1. if we would get client-side Sentry reporting for free, and
2. if we would miss any exceptions.